src/core/qgsmatrix4x4.h

```diff
@@ -0,0 +1,84 @@
+/***************************************************************************
+  qgsmatrix4x4.h
+  --------------------------------------
+  Date                 : July 2023
+  Copyright            : (C) 2023 by Martin Dobias
+  Email                : wonder dot sk at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSMATRIX4X4_H
+#define QGSMATRIX4X4_H
+
+#include "qgis_core.h"
+
+#include "qgsvector3d.h"
+
+#define SIP_NO_FILE
+
+/**
+ * \ingroup core
+ * \brief A simple 4x4 matrix implementation useful for transformation in 3D space.
+ *
+ * It is similar to QMatrix4x4, but working with double precision values.
+ * Most of the time, doing transform using QMatrix4x4 is fine, however QgsMatrix4x4
+ * is useful in situations where single precision floats are not enough.
+ * For example, when using transform matrix where translation component has
+ * values in order of millions.
+ *
+ * \warning Only for internal use
+ *
+ * \note Not available in Python bindings
```

Comment:
why? Later, I guess, it could be useful.

Reply:
It is a low-level class, an implementation detail that is not suited for public API use. We can make it a part of Python API when there will be a cause for it...

Reply:
FWIW, I'm currently leaning toward an "expose to python, but mark as non-stable API" for situations like this. My rationale is:

- it makes test development MUCH quicker (no recompilation necessary), which ultimately results in better test coverage
- it avoids the forced creation of additional .cpp tests , which all add non-negligible recompilation/linking costs
- there's nothing more frustrating when writing a plugin then bumping into a situation where something isn't exposed in PyQGIS, and you either need to re-write a bunch of existing cpp code into Python or are forced to expose the method in QGIS master, wait for release, and put an aggressive minimum QGIS version on the plugin users. While it's not particularly nice to have exposed classes marked as non-stable, I feel it's the lesser of two evils, and with a plugin development hat on I'd rather wear the risk of a non-stable api changing vs being forced to rewrite stuff from QGIS c++ in Python myself.

Reply:
Ok, fair enough, I will expose it...

As for C++/Python tests, I agree with the point of more compilation to do, but on the other hand, debugging PyQGIS tests when they break is more pain :-)

Reply:
Python support added in [2881df9](https://github.com/qgis/QGIS/pull/53842/commits/2881df99f50f472c4d399ec191ef9191e2dac2cd)